flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java

```diff
@@ -186,7 +186,7 @@
 	@Documentation.ExcludeFromDocumentation("dev use only; likely temporary")
 	public static final ConfigOption<Boolean> FORCE_PARTITION_RELEASE_ON_CONSUMPTION =
 			key("jobmanager.scheduler.partition.force-release-on-consumption")
-			.defaultValue(true);
+			.defaultValue(false);
```

Comment:
Quick question: How would Flink behave if failover strategy is set to `full` and this config option being set to `false`? Would we keep blocking result partitions until a pipelined region has been completely executed?

Reply:
the partitions will be released when the vertices are being reset; i.e. on each full restart all partitions would be released.

Reply:
if no failures occur, then yes we will keep partitions until the failover region is consumed.

Long-term we should couple the partition release logic / releasedOnConsumption flag with the failover strategy so we don't keep partitions around for no reason.